src/github.com/getlantern/balancer/balancer.go

```diff
@@ -22,21 +22,30 @@ var (
 // Balancer balances connections established by one or more Dialers.
 type Balancer struct {
 	dialers []*dialer
+	trusted []*dialer
 }
 
 // New creates a new Balancer using the supplied Dialers.
 func New(dialers ...*Dialer) *Balancer {
-	dhs := make([]*dialer, 0, len(dialers))
+	bal := new(Balancer)
+
+	// Sort dialers by QOS (ascending) for later selection
+	sort.Sort(byQOSAscending(dialers))
```

Comment:
It probably doesn't matter here, but in general it's considered bad form to mutate[1] arguments passed in unless that's the very purpose of the function.  The safest sequence would be to copy `dialers` into `bal.dialers`, then sort `bal.dialers` (which would leave `dialers` alone), then iterate over `bal.dialers` to collect the trusted ones into `bal.trusted`.

[1] I wasn't sure whether `...this` declaration syntax would result in the same slice being passed when called with `this...` syntax, or whether a copy would be made.  It turns out the same slice is passed, so the callee can modify it in place, with changes visible to the caller.  Try this: 

```go
package main

import "fmt"

func main() {
	ss := []string{"zero", "one", "two"}
	fmt.Println(ss)
	change(ss...)
	fmt.Println(ss)
}

func change(strs ...string) {
	strs[2] = "whoops"
}
```